main.js

```diff
@@ -1 +1,145 @@
+                } else {
+                    sourceData [caseNo][posY][posX] = 0
+                }
+                readerPlace++;
+            }
+            readerPlace ++;
+        }
+        readerPlace+=2;
+    }
+
+    for (caseNo = 0; caseNo<casesSizes.length; caseNo++) {
+        var printline;
+        onScreen("caseNo:" + (caseNo+1));
+        for (posY=0; posY<casesSizes[caseNo]; posY++) {
+            printline = "";
+            for (posX=0; posX<casesSizes[caseNo]; posX++){
+                printline += sourceData[caseNo][posY][posX];
+            }
+            onScreen(printline);
+        }
+    }
+}
+
+function searcher () {
+    function arrSum (casNo,startX,startY,endX,endY) {
+        var arSum=0;
+        for (y=startY; y<endY;y++) {
+            for (x=startX; x<endX;x++) {
+                arSum += sourceData[casNo][y][x];
+            }
+        }
+        return arSum
+    }
+
+    var endCaseSearch;
+    for (caseNo = 0; caseNo < casesSizes.length; caseNo++) {
+        endCaseSearch = false;
+        for (posY = 0; posY < casesSizes[caseNo]; posY++) {
+            for (posX = 0; posX < casesSizes[caseNo]; posX++) {
+                if (sourceData[caseNo][posY][posX] === 1) {
+                    var startX=posX; startY=posY; sqSize = 2;
+                    if (sourceData[caseNo][posY][posX + 1] === 1) {
+                        for (pY = posY, pX = posX + 2; pX <= (casesSizes[caseNo] - posX); pX++) {
+                            if (sourceData[caseNo][pY][pX] === 1) {
+                                sqSize ++;
+                            }
+                        }
+                        var total = 0;
+                        total = arrSum(caseNo,startX,startY,startX+sqSize,startY+sqSize);
+                        if (total === (sqSize * sqSize)) {
+                            if (casesTotals[caseNo] === total) {
+                                endCaseSearch = true;
+                                casesOutputs[caseNo]="YES";
+                                break;
+                            } else {
+                                endCaseSearch = true;
+                                casesOutputs[caseNo]="NO";
+                                break;
+                            }
+                        } else {
+                            endCaseSearch = true;
+                            casesOutputs[caseNo]="NO";
+                            break
+                        }
+                    } else {
+                        endCaseSearch = true;
+                        casesOutputs[caseNo]="NO";
+                        break;
+                    }
+                }
+                if (endCaseSearch) {
+                    break
+                }
+            }
+            if (endCaseSearch) {
+                break
+            }
+        }
+    }
+
+    for (i=0; i<casesOutputs.length; i++) {
+        onScreen("Case #" + (i + 1) + ": " + casesOutputs[i])
+    }
+}
+
+loadLocalFile();
```

Comment:
LoadLocalFile can be more generic. For example you can pass a filename into it:
loadLocalFile('someFile.txt')